app/src/candid/candid_parser.c

```diff
@@ -158,6 +167,18 @@ static parser_error_t readCandidNat(parser_context_t *ctx, uint64_t *v) {
     return parser_ok;
 }
 
+static parser_error_t readCandidNat32(parser_context_t *ctx, uint32_t *v) {
+    // need to compose it because ledger cannot deference misaligned values
+    uint8_t b;
+    *v = 0;
+
+    for (uint64_t i = 0; i < 32; i += 8) {
+        CHECK_PARSER_ERR(readCandidByte(ctx, &b))
+        *v += (uint64_t) b << i;
```

Comment:
cast as **uint32_t** instead of **uint64_t**

Reply:
good one! It's probably not affecting as we are using `uint8_t` shifted, at max, 3 bytes. Better to have everything controlled 👌🏻